app/components/Views/LedgerSelectAccount/index.tsx

```diff
@@ -0,0 +1,193 @@
+  unlockLedgerWalletAccount,
+} from '../../../core/Ledger/Ledger';
+import LedgerConnect from '../LedgerConnect';
+import { setReloadAccounts } from '../../../actions/accounts';
+import { StackActions, useNavigation } from '@react-navigation/native';
+import { useDispatch } from 'react-redux';
+import { KeyringController } from '@metamask/keyring-controller';
+import { StackNavigationProp } from '@react-navigation/stack';
+import createStyles from './index.styles';
+import OperationTypes from '../../../core/Ledger/types';
+import { HardwareDeviceTypes } from '../../../core/Analytics/MetaMetrics.types';
+
+const LedgerSelectAccount = () => {
+  const navigation = useNavigation<StackNavigationProp<never>>();
+  const dispatch = useDispatch();
+  const { colors } = useTheme();
+  const { trackEvent } = useMetrics();
+  const styles = createStyles(colors);
+  const ledgerThemedImage = useAssetFromTheme(
+    ledgerDeviceLightImage,
+    ledgerDeviceDarkImage,
+  );
+
+  const keyringController = useMemo(() => {
+    const { KeyringController: controller } = Engine.context as {
+      KeyringController: KeyringController;
+    };
+    return controller;
+  }, []);
+
+  const [blockingModalVisible, setBlockingModalVisible] = useState(false);
+  const [accounts, setAccounts] = useState<
+    { address: string; index: number; balance: string }[]
+  >([]);
+
+  const [unlockAccounts, setUnlockAccounts] = useState({
+    trigger: false,
+    accountIndexes: [] as number[],
+  });
+
+  const [forgetDevice, setForgetDevice] = useState(false);
+
+  const [existingAccounts, setExistingAccounts] = useState<string[]>([]);
+
+  useEffect(() => {
+    keyringController.getAccounts().then((value: string[]) => {
+      setExistingAccounts(value);
+    });
+  }, [keyringController]);
+
+  const onConnectHardware = useCallback(async () => {
+    trackEvent(MetaMetricsEvents.CONTINUE_LEDGER_HARDWARE_WALLET, {
+      device_type: HardwareDeviceTypes.LEDGER,
+    });
+    const _accounts = await getLedgerAccountsByOperation(
+      OperationTypes.GET_FIRST_PAGE,
+    );
+    setAccounts(_accounts);
+  }, [trackEvent]);
+
+  const nextPage = useCallback(async () => {
+    const _accounts = await getLedgerAccountsByOperation(
+      OperationTypes.GET_NEXT_PAGE,
+    );
+    setAccounts(_accounts);
+  }, []);
+
+  const prevPage = useCallback(async () => {
+    const _accounts = await getLedgerAccountsByOperation(
+      OperationTypes.GET_PREVIOUS_PAGE,
+    );
+    setAccounts(_accounts);
+  }, []);
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  const onToggle = useCallback(() => {}, []);
+
+  const onUnlock = useCallback(
+    async (accountIndexes: number[]) => {
+      setBlockingModalVisible(true);
+
+      try {
+        for (const index of accountIndexes) {
+          await unlockLedgerWalletAccount(index);
+        }
+      } catch (err) {
+        Logger.log('Error: Connecting Ledger hardware wallet', err);
```

Comment:
Lets omit this, I don't think it's relevant to log it since the connection error could be that the device is not unlocked

```suggestion
        // Do nothing
```